### PR TITLE
7239 log file deletions issued over NFS

### PR DESCRIPTION
**Pull request by dankimmel**
Reviewed by: Sebastien Roy <sebastien.roy@delphix.com>
Reviewed by: George Wilson <george.wilson@delphix.com>

Upstream bugs: DLPX-28790, DLPX-39919

### CODE REVIEW

**Comment by ahrens** [comment]
@zettabot go

File: usr/src/uts/common/fs/nfs/nfs4_srv.c [hunk 1]
```diff
@@ -250,6 +251,8 @@ static nfsstat4 check_open_access(uint32_t,
 				struct compound_state *, struct svc_req *);
 nfsstat4 rfs4_client_sysid(rfs4_client_t *, sysid_t *);
 void rfs4_ss_clid(rfs4_client_t *);
+void rfs_log_deletions(int rfs_version, char *name, struct svc_req *req,
```
Comment (rmustacc) [inline usr/src/uts/common/fs/nfs/nfs4_srv.c]: Stylistically most of the other callers here don't have args in the prototype. Is there a reason this is different?

File: usr/src/uts/common/fs/nfs/nfs4_srv.c [hunk 2]
```diff
@@ -88,6 +88,7 @@ static int rfs4_maxlock_tries = RFS4_MAXLOCK_TRIES;
 static clock_t  rfs4_lock_delay = RFS4_LOCK_DELAY;
 extern struct svc_ops rdma_svc_ops;
 extern int nfs_loaned_buffers;
+extern int log_nfs_deletions;
```
Comment (rmustacc) [inline usr/src/uts/common/fs/nfs/nfs4_srv.c]: This type is a boolean_t in the previous file, here it's an int. They should be consistent types. However, it doesn't seem that this is even used here and it should just be removed.

File: usr/src/uts/common/fs/nfs/nfs3_srv.c [hunk 3]
```diff
@@ -86,6 +86,8 @@ extern int nfs_loaned_buffers;
 
 u_longlong_t nfs3_srv_caller_id;
 
+boolean_t log_nfs_deletions = B_TRUE;
```
Comment (rmustacc) [inline usr/src/uts/common/fs/nfs/nfs3_srv.c]: Why default to B_TRUE? Do most folks actually want remote removals logged?

**Comment by danmcd** [comment]
I'd like @rmustacc's comments addressed before I review this code.

File: usr/src/uts/common/fs/nfs/nfs4_srv.c (hunk 1 above)
Comment (dankimmel) [inline usr/src/uts/common/fs/nfs/nfs4_srv.c]: Nope, consistency seems better here. I'll update this.

File: usr/src/uts/common/fs/nfs/nfs4_srv.c (hunk 2 above)
Comment (dankimmel) [inline usr/src/uts/common/fs/nfs/nfs4_srv.c]: Thanks for catching that, I'll just delete it.

File: usr/src/uts/common/fs/nfs/nfs3_srv.c (hunk 3 above)
Comment (dankimmel) [inline usr/src/uts/common/fs/nfs/nfs3_srv.c]: I'm fine making it disabled by default instead.

**Comment by danmcd** [comment]
Much better from my POV.  Make sure you handle Marcel's mailing-list ones as well, but AFAIC, ship it.

**Comment by dankimmel** [comment]
Closing this PR after feedback from Marcel on the mailing list -- more extensive NFS logging can already be enabled through `nfslogd` if desired, so this is a bit redundant and probably not broadly useful.

**Pull request closed without merging**